basictracer/recorder.py

```diff
@@ -11,6 +13,28 @@ def record_span(self, span):
         pass
 
 
+class InMemoryRecorder(SpanRecorder):
+    """InMemoryRecorder stores all received spans in an internal list.
+
+    This recorder is not suitable for production use, only for testing.
+    """
+    def __init__(self):
+        self.spans = []
+        self.mux = threading.Lock()
+
+    def record_span(self, span):
+        with self.mux:
+            self.spans.append(span)
+
+    def get_spans(self):
+        with self.mux:
+            return self.spans[:]
+
+    def get_sampled_spans(self):
```

Comment:
IMO it would be preferable for callers to just do

```
sampled_spans = filter(lambda s: s.context.sampled, in_mem_rec.get_spans())
```

(or via the list comprehension, whichever)